### PR TITLE
Update eslint-config-airbnb to version 19

### LINKED ISSUE #2836: Update eslint-config-airbnb to version 19
### Description

Update `eslint-config-airbnb` to the latest major version. This contains breaking changes as it drops support for ESLint < 7 and changes various rules.

Also should update `eslint-config-prettier` at the same time.

### PR DESCRIPTION
**Pull request by jpveooys**
## Related issue

Resolves #2836 

## Overview

This updates eslint-config-airbnb to version 19. The new version includes official compatibility with ESLint 8, drops support for ESLint 6 and older and adds a few new rules.

## Link to preview

https://5e25c277526d380020b5e418-chdlredmtk.chromatic.com/

## Reason

Keep linting rules up to date and in line with latest ESLint version.

## Work carried out

- [x] Update eslint-config-airbnb
- [x] Remove ESLint 5 and 6
- [x] Turn off `react/function-component-definition` (see below)
- [x] Make code compliant with new rules

## Developer notes

Would recommend hiding white space in the diff to make it easier to review:

image

Errors were fixed for:

- [react/no-unstable-nested-components](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md)
- [react/jsx-no-constructed-context-values](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-constructed-context-values.md)
- [react/jsx-no-useless-fragment](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md)
- [no-promise-executor-return](https://eslint.org/docs/rules/no-promise-executor-return)
- [default-param-last](https://eslint.org/docs/rules/default-param-last)

[`react/function-component-definition`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md) was turned off as [it's configured as](https://github.com/airbnb/javascript/blob/1bc8cabd447ef9ce69d27ae70e67eb47daeffc21/packages/eslint-config-airbnb/rules/react.js#L528-L531):

```
    'react/function-component-definition': ['error', {
      namedComponents: 'function-expression',
      unnamedComponents: 'function-expression',
    }],
```

which isn't the style we use, and also seems too opinionated for a shared configuration.

### CODE REVIEW

File: packages/react-component-library/src/components/RangeSlider/ThresholdTrack.tsx [hunk 1]
```diff
@@ -15,13 +15,34 @@ export interface ThresholdTrackProps extends TrackItem {
...
 export interface ChunkProps {
+  getTrackProps: GetTrackProps
   $left: number
   $width: number
   $maxWidth: number
   testId: string
   $thresholdColor?: ThresholdColor
 }
 
+const Chunk: React.FC<ChunkProps> = ({
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/RangeSlider/ThresholdTrack.tsx:26]: Although this change is fine for `RangeSliderE` it causes a weird rendering bug in the original `RangeSlider` implementation.

**Review by m7kvqbe1 — COMMENTED** [review]
Nice!

Worth rebasing `release/3.0.0` sooner rather than later when this is merged.

Just left a comment about `RangeSlider` regression.

File: packages/react-component-library/src/components/RangeSlider/ThresholdTrack.tsx (hunk 1 above)
Comment (jpveooys) [reply packages/react-component-library/src/components/RangeSlider/ThresholdTrack.tsx:26]: Have pushed a fix for that: https://5e25c277526d380020b5e418-chdlredmtk.chromatic.com/?path=/docs/range-slider--double-threshold

(Although this is targetting `release/3.0.0` anyway :smile: )

**Review by m7kvqbe1 — APPROVED** [review]

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**